x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -234,23 +251,39 @@ void deleteSnapshots(Map<String, List<SnapshotInfo>> snapshotsToDelete, TimeValu
 
     /**
      * Delete the given snapshot from the repository in blocking manner
+     * @param repo The repository the snapshot is in
+     * @param info The snapshot metadata
+     * @return If present, a SnapshotHistoryItem containing the results of the deletion. Empty if no response or interrupted.
      */
-    void deleteSnapshot(String repo, SnapshotId snapshot) {
-        logger.info("[{}] snapshot retention deleting snapshot [{}]", repo, snapshot);
+    Optional<SnapshotHistoryItem> deleteSnapshot(String repo, SnapshotInfo info) {
+        logger.info("[{}] snapshot retention deleting snapshot [{}]", repo, info.snapshotId());
         CountDownLatch latch = new CountDownLatch(1);
-        client.admin().cluster().prepareDeleteSnapshot(repo, snapshot.getName())
+        String policyId = (String) info.userMetadata().get(POLICY_ID_METADATA_FIELD); // TODO: use getPolicyID once #45362 is merged
+        AtomicReference<SnapshotHistoryItem> result = new AtomicReference<>();
```

Comment:
I think the listener/chained model would be better than putting a result in the `AtomicReference`, what do you think?